File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        out = [1] * len(nums)
        pref = 1
        for i in range(len(nums)):
            out[i] = pref
            pref = nums[i] * pref
        post = 1
        for i in range(len(nums) - 1, -1, -1):
            out[i] = out[i] * post
            post = nums[i] * post
        
        return out

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int last = nums.length-1;
        int k = 0;
        for (int i = 0; i <= last - k; i++) {
            if (nums[i] == val) {
                while (last - k >= 0 && nums[last - k] == val) {
                    k++;
                }
                if (i <= last - k) {
                    nums[i] = nums[last - k];
                    k++;
                }
            }
        }
        return (last+1)-k;
    }
}